File: src/ui/app/root/application/models/Role/RolePrevilages.ts
export interface IRolePrevilages {
    rolePrevilagesLinkId: string;
    userId: number;
    moduleStore: string;


}
export interface IRoles {
    roleId: string;
    parentRoleId: string;
    enabled: boolean;
    name: string;
    createdBy: number;
    createdOn: string;
    lastUpdatedBy?: number;
    lastUpdatedOn?: string;
    roleId1: string;
}


export interface IRoleAssignedList {
    id: string;
    roleId: string;
    storeData: string;
    moduleType: string;
}
export interface ISectionRightLink {
    sectionRightLinkId: string;
    allowedSection: string;
    userId: number;
}

export interface IMaskRightLink {
    maskRightLinkId: string;
    allowedMask: string;
    userId: number;
}
export interface IUserList {

    uid: number;
    name: string;
    type: string;

}

export interface PrevilagesData {
    id: string;
    model: string;
}

export interface ISectionCampusVM {
    sectionCourseLinkId: string;
    campusProgramId: string;
    className: string;
    sectionName: string;
    description: string;
    classId: string;
    sessionId: string;
    campusId: string;
    programDetailId: string;
    sectionId: string;
    isChecked: boolean;
}
export interface RoleDashboard {
    roleDasboardId: number;
    roleId: string;
    moduleId: string;
    columnWidth: string;
    orderBy: number;
}
export interface IRoleDashboardfilter {
    roleDasboardId: number;
    moduleId: string;
    columnWidth: string;
    orderBy: number;

}

export interface UserRoleDashboardlink {
    userDashboadId: string;
    roleDasboardId: number;
    roleId: string;
    moduleId: string;
    columnWidth: string;
    orderBy: number;
    userId: number;
    statusId: number;
}

export interface IOperation{
    link: string
}

export interface TodoList {
    todoListId: string;
    userId: number;
    dated: Date;
    description: string;
    taskStatus: boolean;
    title: string;
    statusId: number;
}

File: src/data/Model/RolePrevilages.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Cms360.Data.Model
{
    [Table("RolePrevilages", Schema = "Role")]
    public partial class RolePrevilages
    {
        [Key]

        public Guid RolePrevilagesLinkId { get; set; }
        public int UserId { get; set; }

        [Column(TypeName = "jsonb")]
        public string ModuleStore { get; set; }





    }

    public class MaskList
    {
        [Key]
        public Guid Id { get; set; }
        public string Text { get; set; }

        public bool IsChecked { get; set; }

    }

    public class Roles
    {
        [Key]
        public string RoleId { get; set; }
        public string ParentRoleId { get; set; }
        public bool Enabled { get; set; }
        public string Name { get; set; }
        public long CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public long? LastUpdatedBy { get; set; }
        public DateTime? LastUpdatedOn { get; set; }

        public string RoleId1 { get; set; }


    }

    public class RoleAssignedList
    {
        [Key]

        public Guid id { get; set; }
        public string RoleId { get; set; }
        public string StoreData { get; set; }
        public string ModuleType { get; set; }



    }


    public class GetRoleUpdateData

    {
        [Key]
        public Guid id { get; set; }
        public string name { get; set; }

        public bool Status { get; set; }
    }

    public class Result
    {
        [Key]

        public int UpdateRolePrevilagesData { get; set; }
    }

    public class UserList
    {
        [Key]
        public int Uid { get; set; }
        public string Name { get; set; }

        public string Type { get; set; }
    }

    public class PrevilagesData
    {
        [Key]
        public string Id { get; set; }
        public string Model { get; set; }
    }

    public class UserRoleDashboardlink
    {
        [Key]
        public Guid UserDashboadId { get; set; }
        public int RoleDasboardId { get; set; }
        public string RoleId { get; set; }

        public string ModuleId { get; set; }
        public string ColumnWidth { get; set; }
        public int OrderBy { get; set; }
        public int UserId { get; set; }
        public int StatusId { get; set; }

    }
}
